tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/security/identity/IdentityAuthorizationService.java

```diff
@@ -102,4 +71,76 @@ public boolean isAllowedToStartProcess(final String processDefinitionKey) {
         getProcessDefinitionsFromAuthorization(),
         Set.of(IdentityProperties.ALL_RESOURCES, processDefinitionKey));
   }
+
+  public List<String> getProcessReadFromAuthorization() {
+    return getFromAuthorization(IdentityAuthorization.PROCESS_PERMISSION_READ);
+  }
+
+  public List<String> getProcessDefinitionsFromAuthorization() {
+    return getFromAuthorization(IdentityAuthorization.PROCESS_PERMISSION_START);
+  }
+
+  private List<String> getFromAuthorization(final String type) {
```

Comment:
**Restructuring suggestion:** It looks like the common pattern here is get an IdentityAuthorization object and then select the correct list of permissions. Could we simplify this to 

```java
@Nullable
private IdentityAuthorization getIdentityAuthorization() {
  if (!tasklistPropertires.getIdentity().isResourcePermissionsEnabled()) {
    return null;
  }
  if (tasklistProperties.getIdentity().getBaseUrl() == null) {
    return null;
  }

  final Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
  return switch (authentication) {
    case IdentityAuthentication identityAuthentication -> identityAuthentication.getAuthorizations()
    case JwtAuthenticationToken jwtAuthenticationToken -> {
      final Identity = SpringContextHolder.getBean(Identity.class);
      yield new IdentityAuthorization(identity.authorizations().forToken(jwtAuthenticationToken.getToken().getTokenValue())
    }
    case TokenAuthentication tokenAuthentication -> {
      final Identity = SpringContextHolder.getBean(Identity.class);
      yield new IdentityAuthorization(
        identity
          .authorizations()
          .forToken(tokenAuthentication.getAccessToken(), tokenAuthentication.getOrganization()))
    }
    default -> null
  }
}

private List<String> getFromAuthorization(final String type) {
  final IdentityAuthorization auth = getIdentityAuthorization();
  if (auth == null) {
    return singletonList(IdentityProperties.ALL_RESOURCES);
  }
  return switch (type) {
    case IdentityAuthorization.PROCESS_PERMISSION_READ -> auth.getProcessAllowedToRead()
    case IdentityAuthorization.PROCESS_PERMISSION_START -> auth.getProcessAllowedToStart()
    default -> emptyList()
  }
}
```

Reply:
@douglasbouttell-camunda I liked it, refactor implemented with minor changes on your suggestion.